File: Tic Toe/Models/Player.ts
import { PlayingPiece } from "./PlayingPiece";

export class Player {
  public name: string;
  public piece: PlayingPiece;

  constructor(name: string, piece: PlayingPiece) {
    this.name = name;
    this.piece = piece;
  }

  setName(name: string) {
    this.name = name;
  }

  setPiece(piece: PlayingPiece) {
    this.piece = piece;
  }

  getName() {
    return this.name;
  }

  getPiece() {
    return this.piece;
  }
}

File: TicToe/Player.java
package TicToe;

public class Player {
    private String name;
    private PlayingPiece piece;
    private int score;

    public Player(String name, PlayingPiece piece) {
        this.name = name;
        this.piece = piece;
        this.score = 0;
    }

    public void incrementScore() {
        this.score++;
    }

    public String getName() {
        return name;
    }

    public int getScore() {
        return score;
    }

    public PlayingPiece getPiece() {
        return piece;
    }
}
